src/cider/tests/utils/CiderNextgenTestBase.cpp

```diff
@@ -39,19 +39,75 @@ void CiderNextgenTestBase::assertQuery(const std::string& sql,
   // As a result, in this case, we need feed a json file, which is delivered by Velox and
   // will be used to generate Substrait plan.
   auto file_or_sql = json_file.size() ? json_file : sql;
-  cider_nextgen_query_runner_->runQueryOneBatch(
-      file_or_sql, *input_array_, *input_schema_, output_array, output_schema);
+  cider_nextgen_query_runner_->runQueryOneBatch(file_or_sql,
+                                                *input_array_,
+                                                *input_schema_,
+                                                output_array,
+                                                output_schema,
+                                                codegen_options_);
   if (0 == duck_res_arrow.size()) {
     // result is empty.
+    CHECK_EQ(0, output_array.length);
   } else {
-    EXPECT_TRUE(CiderArrowChecker::checkArrowEq(duck_res_arrow[0].first.get(),
-                                                &output_array,
-                                                duck_res_arrow[0].second.get(),
-                                                &output_schema));
+    if (ignore_order) {
+      EXPECT_TRUE(
+          CiderArrowChecker::checkArrowEqIgnoreOrder(duck_res_arrow[0].first.get(),
+                                                     &output_array,
+                                                     duck_res_arrow[0].second.get(),
+                                                     &output_schema));
+    } else {
+      EXPECT_TRUE(CiderArrowChecker::checkArrowEq(duck_res_arrow[0].first.get(),
+                                                  &output_array,
+                                                  duck_res_arrow[0].second.get(),
+                                                  &output_schema));
+    }
   }
 
   output_array.release(&output_array);
   output_schema.release(&output_schema);
 }
 
+bool CiderNextgenTestBase::executeIncorrectQuery(const std::string& wrong_sql) {
+  try {
+    struct ArrowArray output_array;
+    struct ArrowSchema output_schema;
+    cider_nextgen_query_runner_->runQueryOneBatch(wrong_sql,
+                                                  *input_array_,
+                                                  *input_schema_,
+                                                  output_array,
+                                                  output_schema,
+                                                  codegen_options_);
+  } catch (const CiderException& e) {
+    LOG(ERROR) << e.what();
+    return true;
```

Comment:
may throw other exception(ciderCompileException), but also pass test here? can return expectTrue(expect_error_code, actual_error_code)?

Reply:
seems need a lot change. we can refactor this in later PR.